tools/airgap_linter.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+#
+# Validates that the supplied framework (identified by the directory passed in) will function, to the best
+# of our knowledge, in an airgapped cluster.
+#
+# The checks are:
+#   - No external URIs defined in anything but resource.json
+#   - No images defined in anything but resource.json (and templated into the svc.yml)
+#
+
+import re
+import sys
+import os
+
+
+def extract_uris(file_name):
+    with open(file_name, "r") as file:
+        lines = file.readlines()
+
+    matcher = re.compile(".*(https:\/\/|http:\/\/)([^\/]*)", re.IGNORECASE)
+    matches = []
+    for line in lines:
+        match = matcher.match(line)
+        if match:
+            matches.append(match.group(2))
+
+    return matches
+
+
+def validate_uri(uri):
+    # URIs are valid if they:
+    #   - end with .mesos
+    #   - contain .dcos
+    if ".dcos" in uri:
+        return True
+
+    if uri.endswith(".mesos"):
+        return True
+
+    return False
+
+
+
+def validate_uris_in(file_name):
+    uris = extract_uris(file_name)
+
+    bad_uri = False
+    for uri in uris:
+        if not validate_uri(uri):
+            print("Found a bad URI:", uri, "in:", file_name)
+            bad_uri = True
+
+    return not bad_uri
+
+
+def get_all_files_to_check(framework_directory):
+    # There's a set of files that will always be present.
+    files = [os.path.join(framework_directory, "universe", "config.json"),
+             os.path.join(framework_directory, "universe", "marathon.json.mustache")]
+
+    # Always check every file in the `dist` directory of the scheduler.
+    dist_dir = os.path.join(framework_directory, "src", "main", "dist")
+
+    for file in os.listdir(dist_dir):
+        files.append(os.path.join(dist_dir, file))
+
+    return files
+
+
+def validate_all_uris(framework_directory):
+    bad_file = False
+    files = get_all_files_to_check(framework_directory)
+    for file in files:
+        if not validate_uris_in(file):
+            bad_file = True
+
+    return not bad_file
+
+
+def validate_images(framework_directory):
+    svc_yml = os.path.join(framework_directory, "src", "main", "dist", "svc.yml")
+
+    with open(svc_yml, "r") as file:
+        lines = file.readlines()
+
+    bad_image = False
+    for line in lines:
+        if "image:" in line:
+            image_matcher = re.compile("\s*image:\s?(.*)", re.IGNORECASE)
```

Comment:
Probably:
```
   image_matcher = re.compile("\s*image:\s?(.*)\s*$", re.IGNORECASE)
```
to avoid trailing whitespace, though i'd probably just, instead

```
    line = line.strip()
    image_matcher = re.compile("image:\s?(.*)$", re.IGNORECASE)
```